src/components/video.tsx

```diff
@@ -109,15 +109,22 @@ export function getVideoProps(allProps: VideoProps, state: { asset?: Asset }) {
 
   if (asset) {
     if (asset.status === 'ready') {
-      props.blurDataURL = blurDataURL ?? asset.blurDataURL;
+      props.blurDataURL ??= asset.blurDataURL;
 
-      // Mux provider logic
+      // Mux provider
       const playbackId = asset.externalIds?.playbackId;
 
       if (playbackId) {
-        props.src ??= `https://stream.mux.com/${playbackId}.m3u8`;
+        // src can't be overridden by the user.
+        props.src = `https://stream.mux.com/${playbackId}.m3u8`;
```

Comment:
I feel like we could do a better job abstracting away the provider in the json. I at least don't see a reason why we can't have a generic "src" or "url" prop in the json that works across providers, rather than building it from the playbackID on player load here. No change needed for now, but I think it'd be good to refactor this so that provider specific logic isn't need here.

Reply:
crossed my mind as well but I think we can't do away with the `playbackId` (or similar) completely because in some cases it's needed. in the default Mux player we set `playbackId` instead of a `src` because Mux player uses it for generating storyboard urls.
https://github.com/muxinc/next-video/blob/1374d9863caa2dfc4ac53c622b2e594cd7868f18/src/components/default-player.tsx#L34-L49

the generated Mux m3u8 above here is only used for custom players